### DISCUSSION TITLE
Redirect every possible subdomains to a single app

### DISCUSSION THREAD

**Discussion by Tobi-De** [caprover/caprover#1509] [Q&A]
Hi, I hope everyone is doing great, my issue is more related to nginx than to caprover itself, but hoping someone has already done something similar, I ask for help here too. 

I have an application running on caprover, let's call it `myapp`. I added a domain name `myapp.com` to `myapp` and enabled https for it and everything works fine. `myapp` is a [saas](https://en.wikipedia.org/wiki/Software_as_a_service) platform and handles different customers via their own unique url e.g. customer1.myapp.com, customer2.myapp.com.

If I add `customer1.myapp.com` domain to `myapp` using caprover *connect new domain* and enbale https for it, it works fine, but you will agree with me that I can't keep doing this for every new customer. I need to update the nginx configuration for my application so that the request to any subdomain of myapp.com is routed to myapp automatically and with ssl enabled for each of them. Any help would be appreciated, thanks.

**Comment by githubsaturn** [comment]
For the wildcard route, you can edit your nginx config. But for enabling SSL, that won't work. So if you need SSL, you have to automate this process using CapRover CLI. It's an advanced process, so make sure you're really familiar with CapRover's API before doing this. 

https://github.com/caprover/caprover-cli#api
